server/resp/src/main/java/org/infinispan/server/resp/ByteBufferUtils.java

```diff
@@ -75,10 +75,16 @@ public static ByteBuf writeLong(Long result, ByteBufPool alloc) {
    }
 
    public static ByteBuf bytesToResult(Collection<byte[]> results, ByteBufPool alloc) {
+      if (results.isEmpty())
+         return stringToByteBuf("*0\r\n", alloc);
+
       int resultBytesSize = 0;
       for (byte[] result: results) {
-         int length = result.length;
-         if (length > 0) {
+         int length;
+         if (result == null) {
+            // $-1
```

Comment:
:+1: Looks like a different bug

Reply:
I believe the method assumed non-null elements. Should I log a JIRA?

Reply:
I guess it is fine. Nothing from the method made me assume it only allowed non null values though. Looks like it would just throw a NPE and we had to rely upon tests to catch it :) Either way no reason for this method not to support a null value.